monkey/monkey_island/cc/repository/mongo_node_repository.py

```diff
@@ -58,18 +69,18 @@ def _upsert_node(self, node: Node):
         except Exception as err:
             raise StorageError(f"{UPSERT_ERROR_MESSAGE}: {err}")
 
-        if result.matched_count != 0 and result.modified_count != 1:
-            raise StorageError(
-                f'Error updating node with source ID "{node.machine_id}": Expected to update 1 '
-                f"node, but {result.modified_count} were updated"
-            )
-
         if result.matched_count == 0 and result.upserted_id is None:
             raise StorageError(
                 f'Error inserting node with source ID "{node.machine_id}": Expected to insert 1 '
                 f"node, but no nodes were inserted"
             )
 
+    def _get_node_by_id(self, node_id: MachineID) -> Optional[Node]:
+        node_dict = self._nodes_collection.find_one(
+            {SRC_FIELD_NAME: node_id}, {MONGO_OBJECT_ID_KEY: False}
+        )
+        return Node(**node_dict) if node_dict else None
```

Comment:
If this function becomes public, it should raise `UnknownRecordError` instead of returning `None`.

Reply:
Raising an error on find queries doesn't make sense. Repository should just return what it found: `None`. Errors should be for indication that it couldn't do the action the caller asked. `UnknownRecordError` makes sense in the delete case, for example, because repository can't delete what's not there, so it' can't complete the operation it was asked to do.